File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        int mid ;
        while(l<=r)
        {
            
            mid = (l+r)/2;
            System.out.println(l +" "+r);
            if(nums[mid]==target)
            {
                return mid;
            }
            if(nums[l]<=nums[mid]) // left part is sorted
            {
                if(nums[mid]>target && nums[l]<=target)
                {
                    r = mid-1;
                }
                else
                {
                    l = mid+1;
                }
            }
            else
            {
                if(nums[mid]<target && target<=nums[r])
                {
                    l = mid+1;
                }
                else
                {
                    r = mid-1;
                }
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp=[]
        if len(nums)>0:
            dp.append(nums[0])
        if len(nums)>1:
            dp.append(max(nums[0],nums[1]))

        for i in range(2,len(nums)):
            dp.append(max(dp[-1],dp[-2]+nums[i]))
        
        return max(dp[-2],dp[-1]) if len(nums)>1  else nums[0]